src/ott/neural/flows/models.py

```diff
@@ -0,0 +1,174 @@
+  dimension ``joint_hidden_dim``.
+
+  Args:
+    output_dim: Dimensionality of the neural vector field.
+    latent_embed_dim: Dimensionality of the embedding of the data.
+    condition_dim: Dimensionality of the conditioning vector.
+    condition_embed_dim: Dimensionality of the embedding of the condition.
+      If :obj:`None`, set to ``latent_embed_dim``.
+    t_embed_dim: Dimensionality of the time embedding.
+      If :obj:`None`, set to ``latent_embed_dim``.
+    joint_hidden_dim: Dimensionality of the hidden layers of the joint network.
+      If :obj:`None`, set to ``latent_embed_dim + condition_embed_dim +
+      t_embed_dim``.
+    num_layers_per_block: Number of layers per block.
+    act_fn: Activation function.
+    n_frequencies: Number of frequencies to use for the time embedding.
+
+  """
+  output_dim: int
+  latent_embed_dim: int
+  condition_dim: int = 0
+  condition_embed_dim: Optional[int] = None
+  t_embed_dim: Optional[int] = None
+  joint_hidden_dim: Optional[int] = None
+  num_layers_per_block: int = 3
+  act_fn: Callable[[jnp.ndarray], jnp.ndarray] = nn.silu
+  n_frequencies: int = 128
+
+  def __post_init__(self):
+
+    # set embedded dim from latent embedded dim
+    if self.condition_embed_dim is None:
+      self.condition_embed_dim = self.latent_embed_dim
+    if self.t_embed_dim is None:
+      self.t_embed_dim = self.latent_embed_dim
+
+    # set joint hidden dim from all embedded dim
+    concat_embed_dim = (
+        self.latent_embed_dim + self.condition_embed_dim + self.t_embed_dim
+    )
+    if self.joint_hidden_dim is not None:
+      assert (self.joint_hidden_dim >= concat_embed_dim), (
+          "joint_hidden_dim must be greater than or equal to the sum of "
+          "all embedded dimensions. "
+      )
+      self.joint_hidden_dim = self.latent_embed_dim
+    else:
+      self.joint_hidden_dim = concat_embed_dim
+    super().__post_init__()
+
+  @nn.compact
+  def __call__(
+      self,
+      t: jnp.ndarray,
+      x: jnp.ndarray,
+      condition: Optional[jnp.ndarray] = None,
+      rng: Optional[jnp.ndarray] = None,
+  ) -> jnp.ndarray:
+    """Forward pass through the neural vector field.
+
+    Args:
+      t: Time of shape (batch_size, 1).
+      x: Data of shape (batch_size, output_dim).
+      condition: Conditioning vector.
+      rng: Random number generator.
+
+    Returns:
+      Output of the neural vector field.
+    """
+    t = flow_layers.CyclicalTimeEncoder(n_frequencies=self.n_frequencies)(t)
+    t_layer = layers.MLPBlock(
+        dim=self.t_embed_dim,
+        out_dim=self.t_embed_dim,
+        num_layers=self.num_layers_per_block,
+        act_fn=self.act_fn
+    )
+    t = t_layer(t)
+
+    x_layer = layers.MLPBlock(
+        dim=self.latent_embed_dim,
+        out_dim=self.latent_embed_dim,
+        num_layers=self.num_layers_per_block,
+        act_fn=self.act_fn
+    )
+    x = x_layer(x)
+
+    if self.condition_dim > 0:
```

Comment:
I had to revert it, as otherwise the `init/create_train_state` method doesn't work.